File: Python/Program/Smartphone.py
from Komputer import Komputer
from Portable import Portable

class Smartphone(Komputer, Portable):
    # Constructor
    def __init__(self, nama="", cpu=None, gpu=None, storage=None, camera_MP=0, os_phone="", battery_capacity=0, weight=0.0):
        Komputer.__init__(self, nama, cpu, gpu, storage)
        Portable.__init__(self, battery_capacity, weight)
        self.camera_MP = camera_MP
        self.os_phone = os_phone
    
    # Getters
    def get_camera_MP(self):
        return self.camera_MP
    
    def get_os_phone(self):
        return self.os_phone
    
    # Setters
    def set_camera_MP(self, camera_MP):
        self.camera_MP = camera_MP
    
    def set_os_phone(self, os_phone):
        self.os_phone = os_phone

File: Java/Program/Smartphone.java
package Java.Program;

public class Smartphone extends Komputer {
    // Attributes
    private Portable portable;
    private int cameraMP;
    private String osPhone;
    
    // Constructors
    public Smartphone() {
        this.portable = new Portable();
        this.cameraMP = 0;
        this.osPhone = "";
    }
    
    public Smartphone(String nama, Cpu cpu, GPU gpu, Harddrive storage, int cameraMP, String osPhone, int batteryCapacity, float weight) {
        super(nama, cpu, gpu, storage);
        this.portable = new Portable(batteryCapacity, weight);
        this.cameraMP = cameraMP;
        this.osPhone = osPhone;
    }
    
    // Getters
    public int getBatteryCapacity() {
        return this.portable.getBatteryCapacity();
    }
    
    public float getWeight() {
        return this.portable.getWeight();
    }
    
    public int getCameraMP() {
        return this.cameraMP;
    }
    
    public String getOsPhone() {
        return this.osPhone;
    }
    
    // Setters
    public void setBatteryCapacity(int batteryCapacity) {
        this.portable.setBatteryCapacity(batteryCapacity);
    }
    
    public void setWeight(float weight) {
        this.portable.setWeight(weight);
    }
    
    public void setCameraMP(int cameraMP) {
        this.cameraMP = cameraMP;
    }
    
    public void setOsPhone(String osPhone) {
        this.osPhone = osPhone;
    }
}
